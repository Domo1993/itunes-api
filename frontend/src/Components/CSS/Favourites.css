.favHead {
    text-align: center;
    background: lightgray;
    padding-bottom: 30px;
    margin-bottom: 2%;
}

#favHead {
    font-family: 'Nova Flat';
    font-size: 5vw;
    color: rgb(255, 196, 0);
    padding-top: 20px;
    text-shadow: 3px 3px 3px black;
}

#play {
    float: left;
    max-width: 11%;
    height: auto;
    margin-left: 20%;
    margin-top: 2%;
}

#subtitle {
    
}

#link {
    text-align: right;
    text-decoration: none;
    background-color: rgba(24, 23, 109, 0.637);
    color: white;
    text-shadow: 1px 1px 1px black;
    padding: 10px;
}

#link:hover {
    text-decoration: none;
    color: black;
    text-shadow: 1px 1px 1px white;
}

.fas, .fa-trash-alt {
    font-size: 20px;
}

.tableData {
    text-align: center;
}

#link {
    text-decoration: none;
}