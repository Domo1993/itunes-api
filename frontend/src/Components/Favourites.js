import React, {Component} from 'react';
import './CSS/Favourites.css';
import trash from './Images/delete.png';
import parse from 'html-react-parser';					// cleaning the json files text's readability
// Bootstrap imports
import DescriptionModal from './DescriptionModal.js';
import VideoModal from './VideoModal.js';
import { NavLink } from 'react-router-dom';

class Favourites extends Component {
	constructor(props){
		super(props);
		this.state = {
			favourites: []
		}
	}
	// Initial load of the json data from favourites path and stored in the favourites array state
	componentDidMount(){
		fetch('/favourites',{ method: "GET", headers: { "Content-Type": "application/json" } })
		.then(res => res.json())
		.then(favourites => this.setState({ favourites }));
	}
	// This function calls the delete method upon clicking the bin icon
	myDelete = (favId,wrapper,preUrl) => {
		fetch(`/favourites?id=${favId}&wrapperType=${wrapper}&previewUrl=${preUrl}`,{ method: "DELETE", headers: { "Content-Type": "application/json" } })
		.then(() => {
			fetch('/favourites', { method: "GET", headers: { "Content-Type": "application/json" } })
				.then(res => res.json())
				.then(favourites => this.setState({ favourites }));
			})
	}

	// This function will return either an audio tag or video tag(within a modal) depending on the user's search
	mediaCheck = (media, other, source) => {
		if(media === "song" || media === "podcast" || other === "audiobook"){
			return <audio src={source} controls/>
		} else if(media === "music-video" || media === "feature-movie" || media === "tv-episode"){
		return <VideoModal video={<video width="320" height="240" src={(source)} controls> Sorry, your browser doesn't support HTML5</video>} />
		}
	}

	// This function returns the description (within a modal) of an audio book if the user searches audiobook
	audioBookDescripFav = (wrapper,descrip) => {
        if(wrapper === "audiobook"){
            return <DescriptionModal description={parse(`${descrip}`)} />
        }        
    }

	// This function returns the title of the user's searched item/s
    audioBookTitleCheckFav = (wrapper,collectName,songName) => {
        if(wrapper === "audiobook"){
            return <h4>{collectName}</h4>
        } else {
            return <h4>{songName}</h4>
        }
    }

	render(){
		return(
		<div>
			<header className="favHead">
				<h1 id="favHead">Your Favourites</h1>
				<NavLink id="link" to="/">Home</NavLink>
			</header>

			<table>
				{/*Mapping all the favourites data into a table*/}
				{this.state.favourites.map((obj, key) =>{
					return(
					<tbody key={key++}>
						<tr>
							<td className="tableData"><img src={obj.image} alt="cover pic"/><br/>
								{this.audioBookTitleCheckFav(obj.wrapperType, obj.collectionName, obj.title)}
								<p>{obj.artist}</p>
							</td>
							
							<td>{this.mediaCheck(obj.kind, obj.wrapperType, obj.previewUrl)}
							<br/>
								{this.audioBookDescripFav(obj.wrapperType, obj.description)}
							</td>
							<td><button onClick={() => this.myDelete(obj.id,obj.wrapperType,obj.previewUrl)}><img src={trash} alt="trash can"/></button></td>
						</tr>
					</tbody>
					)
				})}
			</table>
		</div>
		)
	}
}

export default Favourites;
