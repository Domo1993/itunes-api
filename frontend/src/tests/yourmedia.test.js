import React from 'react';
import Favourites from '../Components/Favourites';
import renderer from 'react-test-renderer';

test('renders correctly', () => {
const tree = renderer
.create(<Favourites/>)
.toJSON();
expect(tree).toMatchSnapshot();
});