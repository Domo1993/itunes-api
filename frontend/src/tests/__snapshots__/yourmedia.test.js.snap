// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <header
    className="favHead"
  >
    <h1
      id="favHead"
    >
      Your Favourites
    </h1>
    <a
      href="/"
      id="link"
    >
      Home
    </a>
  </header>
  <table />
</div>
`;
